### DISCUSSION TITLE
Passing JS `new` operator to Brython !?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2213] [Q&A] [answered]
Hi everyone,

I would like to call a [JS library](https://leafletjs.com/) trough Brython.

in my previous JS code I had
```JS
var IcoExt = L.Icon.extend({
    options: {
    shadowUrl: 'shadow.png',
    iconSize:     [32, 32],
    shadowSize:   [0, 0],
    iconAnchor:   [16, 31],
    shadowAnchor: [4, 62],
    popupAnchor:  [0, -30]
       }
    });

var pink = new IcoExt({iconUrl: 'Pink.png'});
```
So I don't really understand the last line `var pink = new POI...`
I believe it create an instance of `POIcons` and push `{iconUrl: 'Pink.png'}` into the dict `options` ?
not clear.., I've diged on www but found nothing conclusive.

The first part is easy to translate to Brython/Python
```python
IcoExt = window.L.Icon.extend({
    'options': {
        'shadowUrl': 'static/map/trans.png',
        'iconSize':     [32, 32],
        'shadowSize':   [0, 0],
        'iconAnchor':   [16, 31],
        'shadowAnchor': [4, 62],
        'popupAnchor':  [0, -30]}
        })
```

But the second (related to `new` operator) doesn't work

```python
pink = IcoExt({'iconUrl': 'Pink.png'})
```
give me: 
> Exception: <Javascript TypeError>: can't convert undefined to object

Any ideas ?

Thanks.

**Comment by PierreQuentel** [comment, marked as answer]
A search on "Javascript new" returns [this page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new). 

Brython documentation about [using Javascript objects](https://www.brython.info/static_doc/en/jsobjects.html) explains that Javascript constructors used in Brython code have a method `jsobj.new(args)` which is the equivalent of Javascript `new jsobj(args)`.
